### PR TITLE
Fix root detection on btrfs in rescue mode

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Mount btrfs volumes during rescue mode (to get list of its
subvolumes) and include non-leaf btrfs subvolumes when looking for
root devices and make sure it's mounted with subvol mount option.

Resolves: rhbz#1250011

### CODE REVIEW

File: blivet/__init__.py [hunk 1]
```diff
@@ -2106,7 +2106,7 @@ def mountExistingSystem(fsset, rootDevice,
         rootDevice.setup()
         rootDevice.format.mount(chroot=rootPath,
                                 mountpoint="/",
-                                options=readOnly)
+                                options="%s,%s" % (rootDevice.format.options, readOnly))
```
Comment (vpodzime) [inline blivet/__init__.py]: This should go into a separate commit.

Comment (vojtechtrefny) [inline blivet/__init__.py]: Makes sense, fixed.

**Comment by dwlehman** [comment]
Should we be enabling `installer_mode` for rescue mode? Also, I think you could use the `direct` attribute instead of looking for leaves, &c.

**Comment by vojtechtrefny** [comment]
> Should we be enabling `installer_mode` for rescue mode?

I'm not sure. `installer_mode` enables a lot things including running fsck on filesystems and enabling iscsi and fcoe and I don't know if we want this in rescue mode. I think this solution is safer for rhel7-branch.

> Also, I think you could use the `direct` attribute instead of looking for leaves, &c.

I didn't know we have such attribute :) I changed the code to use `direct` instead of `devicetree.leaves`.

File: blivet/__init__.py [hunk 2]
```diff
@@ -3025,7 +3025,10 @@ def _findExistingInstallations(devicetree):
         util.makedirs(getTargetPhysicalRoot())
 
     roots = []
-    for device in devicetree.leaves:
+
+    direct_devices = [dev for dev in devicetree.devices if dev.direct]
```
Comment (vpodzime) [inline blivet/__init__.py]: This could be a generator.

**Comment by vpodzime** [comment]
Looks good to me other than the nitpick above.
